src/dsk/alloc.c

```diff
@@ -0,0 +1,34 @@
+#include "alloc.h"
+#include "mdisk.h"
+#include "../mkfs/metaBlocks.h"
+
+static pthread_mutex_t iNodeListMutex = PTHREAD_MUTEX_INITIALIZER;
+const int SLEEP_TIME_IN_SECONDS = 3;
+
+size_t alloc(superBlock* superBlock) {
+    while (superBlock -> free_blklst_lock)
+        sleep(SLEEP_TIME_IN_SECONDS);
```

Comment:
This should not be done. If 10 requests come at the same time, one passes through and the remaining sleep for 3 seconds. But the disk read doesn't take 3 seconds. And all the processes wake up after 3 seconds and they contest again, letting only one pass through. This will keep 8 processes to sleep again. This way, the entire thing can take about 30 seconds to finish.

So, we would want only the mutex lock. That way, kernel takes care of which thread must be sleeping and when it must be woken up.